instance/management/commands/instance_statistics_csv.py

```diff
@@ -0,0 +1,294 @@
+        parser.add_argument(
+            '--end-date',
+            default=datetime.utcnow().date().strftime("%Y-%m-%d"),
+            type=valid_date,
+            help='The last day on which statistics should be gathered. '
+            'Defaults to today (UTC).'
+            'FORMAT: YYYY-MM-DD'
+        )
+        parser.add_argument(
+            '--out',
+            default=None,
+            help='Path to the output file of the new CSV. Leave blank to use stdout.'
+        )
+
+    def handle(self, *args, **options):
+        # Determine the stream to be used for outputting the CSV.
+        if options['out'] is None:
+            out = self.stdout
+        else:
+            try:
+                out = open(options['out'], 'w')
+            except PermissionError:
+                self.stderr.write(self.style.ERROR(
+                    'Permission denied while attempting to write '
+                    'file: {outfile}'.format(
+                        outfile=options['out']
+                    )
+                ))
+                sys.exit(1)
+
+        start_date = datetime.strptime(options['start_date'], "%Y-%m-%d").date()
+        end_date = datetime.strptime(options['end_date'], "%Y-%m-%d").date()
+
+        # Verify that --end-date is greater than or equal to --start-date
+        if end_date < start_date:
+            self.stderr.write(self.style.ERROR(
+                '--end-date ({end_date}) must be later than or equal '
+                'to --start-date ({start_date})'.format(
+                    end_date=end_date,
+                    start_date=start_date
+                )
+            ))
+            sys.exit(1)
+
+        self.collect_instance_statistics(out, options['domain'], start_date, end_date)
+
+    def get_instance_from_domain_name(self, domain_name):
+        """ Get an instance object for a given domain name """
+        try:
+            instance = OpenEdXInstance.objects.get(
+                Q(external_lms_domain=domain_name) | Q(internal_lms_domain=domain_name)
+            )
+        except OpenEdXInstance.DoesNotExist:
+            self.stderr.write(self.style.ERROR(
+                'No OpenEdXInstance exists with an external or internal '
+                'domain of {domain_name}'.format(
+                    domain_name=domain_name
+                )
+            ))
+            sys.exit(1)
+
+        # If there are no active appservers for the instance, we should error out
+        if not instance.successfully_provisioned or not instance.get_active_appservers():
+            self.stderr.write(self.style.ERROR(
+                'No active OpenEdXAppServers exist for the instance with '
+                'external or internal domain of {domain_name}'.format(
+                    domain_name=domain_name
+                )
+            ))
+            sys.exit(1)
+
+        return instance
+
+    def get_elasticsearch_statistics(self, playbook_output_dir, name_prefix, start_date, end_date):
```

Comment:
@pcockwell, how about this?

```suggestion
    def get_hits_data_summary_from_elasticsearch(self, playbook_output_dir, name_prefix, start_date, end_date):
```

Reply:
Using `get_elasticsearch_hits_data_summary`. Done in next commit